exonum-java-binding-qa-service/src/main/java/com/exonum/binding/qaservice/transactions/TransactionPreconditions.java

```diff
@@ -18,41 +18,38 @@
 
 import static com.google.common.base.Preconditions.checkArgument;
 
-import com.exonum.binding.common.message.Message;
 import com.exonum.binding.qaservice.PromoteToCore;
 import com.exonum.binding.qaservice.QaService;
+import com.exonum.binding.transaction.RawTransaction;
 
 @PromoteToCore("You have to check these preconditions all the time!")
 final class TransactionPreconditions {
 
   private static final short SERVICE_ID = QaService.ID;
 
-  static <MessageT extends Message> MessageT checkTransaction(MessageT message,
-                                                              short expectedTxId) {
-    short serviceId = message.getServiceId();
+  static <T extends RawTransaction> T checkTransaction(T transaction, short expectedTxId) {
```

Comment:
For the same reasons as in crypto, I'd simplify this to a non-parameterized method.